NeosModLoader/AssemblyHider.cs

```diff
@@ -0,0 +1,72 @@
+using BaseX;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NeosModLoader
+{
+	internal static class AssemblyHider
+	{
+		private static HashSet<Assembly>? neosAssemblies;
+		private static HashSet<Assembly>? modAssemblies;
+
+		/// <summary>
+		/// Patch Neos's type lookup code to not see mod-related types. This is needed, because users can pass
+		/// arbitrary strings to TypeHelper.FindType(), which can be used to detect if someone is running mods.
+		/// </summary>
+		/// <param name="harmony">Our NML harmony instance</param>
+		/// <param name="initialAssemblies">Assemblies that were loaded when NML first started</param>
+		internal static void PatchNeos(Harmony harmony, HashSet<Assembly> initialAssemblies)
+		{
+			if (ModLoaderConfiguration.Get().HideModTypes)
+			{
+				neosAssemblies = GetNeosAssemblies(initialAssemblies);
+				modAssemblies = GetModAssemblies();
+				MethodInfo target = AccessTools.DeclaredMethod(typeof(TypeHelper), nameof(TypeHelper.FindType));
+				MethodInfo patch = AccessTools.DeclaredMethod(typeof(AssemblyHider), nameof(FindTypePostfix));
+				harmony.Patch(target, postfix: new HarmonyMethod(patch));
+			}
+		}
+
+		private static HashSet<Assembly> GetNeosAssemblies(HashSet<Assembly> initialAssemblies)
+		{
+			initialAssemblies.Remove(Assembly.GetExecutingAssembly());
+			initialAssemblies.Remove(typeof(Harmony).Assembly);
```

Comment:
should probably add a comment here clarifying why a specific case is needed.
in theory, if nml is the only thing a user is using that loads harmony, and the user followed the install instructions of placing harmony in nml_libs, initialAssemblies shouldn't contain harmony.
i could be wrong. just sayin its worth commenting.

Reply:
Comment added in f9c6f22d. The install instructions weren't always to use nml_libs, and I don't want to punish users who followed the old steps.